flang/lib/Optimizer/Transforms/DebugTypeGenerator.cpp

```diff
@@ -205,16 +205,38 @@ mlir::LLVM::DITypeAttr DebugTypeGenerator::convertCharacterType(
   if (charTy.getFKind() != 1)
     encoding = llvm::dwarf::DW_ATE_UCS;
 
+  uint64_t sizeInBits = 0;
+  mlir::LLVM::DIExpressionAttr lenExpr = nullptr;
+  mlir::LLVM::DIExpressionAttr locExpr = nullptr;
+
+  if (allocatable) {
+    llvm::SmallVector<mlir::LLVM::DIExpressionElemAttr> ops;
+    auto addOp = [&](unsigned opc, llvm::ArrayRef<uint64_t> vals) {
+      ops.push_back(mlir::LLVM::DIExpressionElemAttr::get(context, opc, vals));
```

Comment:
ultra nit: this might be very slightly faster, feel free to ignore if you feel this harms readability
```suggestion
      ops.emplace_back(context, opc, vals);
```

Reply:
This look concise. I tired to use it but it gives build error. It may be that we are calling `DIExpressionElemAttr::get` and not its constructor so `emplace_back` would not work.